File: blog/php-vs-go/object/object.go
/*
* @Author: GeekWho
* @Date:   2019-01-24 15:45:17
* @Last Modified by:   GeekWho
* @Last Modified time: 2019-01-24 15:45:22
*/
package main

import (
    "fmt"
    "math"
)
// 申明一个结构体
type HelloWorld struct {
    X, Y float64
}
// 添加一个Run方法，首字母大写为公用方法，小写为私有方法。
func (v HelloWorld) Run() float64 {
    return math.Sqrt(v.X*v.X + v.Y*v.Y)
}

func main() {
    // 初始化类
    v := HelloWorld{3, 4}
    fmt.Println(v.Run())
}

File: blog/php-vs-go/object/object.php
<?php

/**
 * @Author: GeekWho
 * @Date:   2019-01-24 15:44:38
 * @Last Modified by:   GeekWho
 * @Last Modified time: 2019-01-24 15:44:48
 */
class HelloWorld
{
    public $x;
    public $y;
    public function __construct($x , $y){
        $this->x = $x;
        $this->y = $y;
    }
    public function run()
    {
        echo sqrt($this->x * $this->x + $this->y * $this->y);
    }
}
(new HelloWorld(3,4))->run();
